src/utils/index.ts

```diff
@@ -141,8 +141,11 @@ export const getCurrentBranch = () => {
 // This could be running on the main branch or on a branch that was created from the main branch.
 // If this is running on the main branch, we want to get all commits since the last release.
 // If this is running on a branch that was created from the main branch, we want to get all commits since the branch was created.
-export const getCommitsSinceRef = (branch: string) => {
-  gitFetch(branch);
+export const getCommitsSinceRef = (branch: string, options: MeowOptions) => {
+  if (options.flags.gitFetch) {
```

Comment:
This feels like kind of a breaking change. As this function is now only works when the flag is provided, right?

Reply:
Haha, I saw this comming - and yeah, it is, *but* - we're still within `0.x.x` release line (comparable to beta?), which I think is common to have breaking changes without `major` bumps - correct me if I'm wrong, I didn't recheck for it now, because we've clean option(s) also.

As an idea, to resolve it clean, we might introduce a `config` using [c12](https://github.com/unjs/c12/) to also already support the seemingly future proof [config-dir](https://github.com/pi0/config-dir/)-convention and make `git-fetch` configurable there, with default of `true` (to `git-fetch`) - same file could also be used to store last `commits`/`hashes` mentioned.

So if someone wants a `default` of `false`, he'll just set this in the config.

Still have to decide which behaviour should be the default, where I would prefer not to `git-fetch`, just because it's a comparable heavy (time/delay) process - but with the config I'm totally fine either way, and then the command line option becomes also negatable, of course.

Just a rough outline so far, but what do you think?

Reply:
My suggestion would be to keep the default as `true` and if people would like to skip it let them pass the flag explicitly. Most of the people run with `npx` (or anyother flavor from package manager they use) without locking the package version so when they get the latest this change maybe an issue. 

Let's update the Readme with the details about it.

What do you think.

Reply:
I still like the additional config options way, but can add this later - so we'll've this one just quickly in, right - will go for it, yeah.